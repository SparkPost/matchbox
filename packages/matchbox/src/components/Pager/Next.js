import React from 'react';
import PropTypes from 'prop-types';
import { ArrowForward } from '@sparkpost/matchbox-icons';
import { Button } from '../Button';
import styles from './Pager.module.scss';

const Next = ({ disabled, onClick }) => (
  <Button className={styles.Next} disabled={disabled} onClick={onClick}>
    <ArrowForward size={16} />
  </Button>
);

Next.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func
};

Next.displayName = 'Pager.Next';

export default Next;
