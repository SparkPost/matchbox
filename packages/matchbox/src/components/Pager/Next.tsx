import React from 'react';
import { ArrowForward } from '@sparkpost/matchbox-icons';
import { Button, PolymorphicButton } from '../Button';
import { ScreenReaderOnly } from '../ScreenReaderOnly';

const Next = React.forwardRef(function Next(props, userRef) {
  return (
    <Button
      size="small"
      width="600"
      {...props}
      ml={props.ml || props.marginRight || 200}
      ref={userRef}
    >
      <ArrowForward size={16} />
      <ScreenReaderOnly>Next</ScreenReaderOnly>
    </Button>
  );
}) as PolymorphicButton;

Next.displayName = 'Pager.Next';
export default Next;
