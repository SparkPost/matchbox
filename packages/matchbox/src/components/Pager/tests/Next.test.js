import React from 'react';
import { shallow } from 'enzyme';
import Next from '../Next';

describe('Pager.Next', () => {
  it('renders button', () => {
    const wrapper = shallow(<Next />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders disabled button', () => {
    const wrapper = shallow(<Next disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls click handler', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Next onClick={handleClick} />);
    wrapper.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
