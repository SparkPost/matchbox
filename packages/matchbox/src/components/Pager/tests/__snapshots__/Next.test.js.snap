// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager.Next renders button 1`] = `
<Button
  className="Next"
  size="default"
>
  <ArrowForward
    size={16}
  />
</Button>
`;

exports[`Pager.Next renders disabled button 1`] = `
<Button
  className="Next"
  disabled={true}
  size="default"
>
  <ArrowForward
    size={16}
  />
</Button>
`;
